src/screens/PaymentRequestDetail.js

```diff
@@ -42,16 +43,22 @@ class PaymentRequestDetail extends React.Component {
   constructor(props) {
     super(props);
 
+    this.state = {
+      paymentIsConfirmed: false,
+    };
+
     this.modalConfirmation = React.createRef();
   }
 
   async componentDidMount() {
-    // When we create a new payment request we update the address for a new one
-    await this.props.wallet.getNextAddress();
+    // When we create a new payment request, we don't update the address for a new one
+    // This will only happen when it receives a transaction and becomes a used address
+    await this.props.wallet.getCurrentAddress();
```

Comment:
I don't understand what's going on here, did we change the logic for `getCurrentAddress` and `getNextAddress`? When?

Reply:
This was requested on item n⁰ 9 on issue 318.

`getNextAddress()` not only fetches the next available address, but also marks the current one as used:

https://github.com/HathorNetwork/hathor-wallet-lib/blob/master/src/new/wallet.js#L563-L565

By changing the call to `getCurrentAddress()` we don't do this marking, and leave it to the wallet's new transaction websocket handler to do so.